src/AppInstallerCLICore/Workflows/WorkflowBase.h

```diff
@@ -216,13 +229,13 @@ namespace AppInstaller::CLI::Workflow
     // Outputs: None
     struct EnsureMatchesFromSearchResult : public WorkflowTask
     {
-        EnsureMatchesFromSearchResult(bool isFromInstalledSource) :
-            WorkflowTask("EnsureMatchesFromSearchResult"), m_isFromInstalledSource(isFromInstalledSource) {}
+        EnsureMatchesFromSearchResult(SearchPurpose purpose) :
+            WorkflowTask("EnsureMatchesFromSearchResult"), m_searchPurpose(purpose) {}

```

Comment:
Check expected searchPurpose upfront unless this function can handle all of them

THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), m_searchPurpose != ...);

Reply:
This can handle all of them, and defaults to `Resource::String::NoPackageFound` in the event new enum values are added